db/migrate/20200129172534_add_epoch_version_release_arch_to_katello_installed_packages.rb

```diff
@@ -0,0 +1,44 @@
+class AddEpochVersionReleaseArchToKatelloInstalledPackages < ActiveRecord::Migration[5.2]
+  def up
+    add_column :katello_installed_packages, :nvrea, :string
+    add_column :katello_installed_packages, :epoch, :string
+    add_column :katello_installed_packages, :version, :string
+    add_column :katello_installed_packages, :release, :string
+    add_column :katello_installed_packages, :arch, :string
+
+    epoch_non_0 = ::Katello::Rpm.where.not(epoch: 0)
+
+    ::Katello::InstalledPackage.reset_column_information
+    ::Katello::InstalledPackage.all.each do |pkg|
+      epoch = epoch_non_0.where(nvra: pkg.nvra).first
+      if epoch.nil?
+        epoch = '0'
+      end
```

Comment:
i think you could do this above to accomplish the same thing: 
`Katello::Rpm.where.not(:epoch => [0, nil]`

Reply:
actually i misunderstood what this was doing, i see now that its looping over every installed package to update it.  I did some testing on a user's db and this seemed really slow.  Something like this:

```
epoch_non_0 = ::Katello::Rpm.where.not(epoch: 0)
installed_packages = Katello::InstalledPackage.where(:nvra => epoch_non_0.select(:nvra)).each do |pkg|
   #DO STUFF 
end
Katello::InstalledPackage.where(:epoch => nil).update_all(:epoch => 0)
```

was much faster.  Here's some stats from the db:

irb(main):077:0> Katello::Rpm.count
=> 198944
irb(main):078:0> Katello::InstalledPackage.count
=> 381726
> installed_packages = Katello::InstalledPackage.where(:nvra => epoch_non_0.select(:nvra)).count
=> 3068

without any of the .save! or update_all  it finished in about ~100 seconds which seems resonable for  a large dataset.

Reply:
@jlsherrill  If I filter the packages using `Katello::InstalledPackage.where(:nvra => epoch_non_0.select(:nvra))`, then only `InstalledPackage`s with a matching `Rpm` will have the `nvrea`, `version`, `release`, and `arch` filled in.  That's why I needed to loop through each individual `InstalledPackage`s.  If you're okay with those fields being `nil` then I can switch to the strategy you mentioned.